crates/editor/src/element.rs

```diff
@@ -9635,11 +9677,15 @@ impl EditorLayout {
     }
 }
 
-struct LineNumberLayout {
+struct LineNumberSegment {
     shaped_line: ShapedLine,
     hitbox: Option<Hitbox>,
 }
 
+struct LineNumberLayout {
+    segments: Vec<LineNumberSegment>,
+}
+
```

Comment:
I believe this is why we were seeing only one line number per logical line. Because each logical line would get a single shaped line and hitbox. I'd love for you to tell me that there's another way, though, and that I'm just missing something.